From a136cbae8f8848d49976512f7f0ec73137428ea0 Mon Sep 17 00:00:00 2001
From: Michael Catanzaro <mcatanzaro@igalia.com>
Date: Thu, 29 Nov 2018 15:34:17 -0600
Subject: [PATCH 01/15] filechoosernative: forward current_filter to delegate
 dialog

gtk_file_chooser_set_filter() doesn't work for GtkFileChooserNative. The
code forwards added and removed filters to the delegate dialog, but
doesn't do anything to set the selected one, so the wrong one gets
chosen. So fix that.

This only fixes the fallback dialog. The portal will be fixed in a
subsequent commit.

Partial fix for #1492
---
 gtk/gtkfilechoosernative.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/gtk/gtkfilechoosernative.c b/gtk/gtkfilechoosernative.c
index cfa274f4f1..cf1d97b31d 100644
--- a/gtk/gtkfilechoosernative.c
+++ b/gtk/gtkfilechoosernative.c
@@ -466,6 +466,7 @@ gtk_file_chooser_native_set_property (GObject      *object,
 
     case GTK_FILE_CHOOSER_PROP_FILTER:
       self->current_filter = g_value_get_object (value);
+      gtk_file_chooser_set_filter (GTK_FILE_CHOOSER (self->dialog), self->current_filter);
       g_object_notify (G_OBJECT (self), "filter");
       break;
 
-- 
2.17.0

